src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsTests/Modals/TestModal.tsx

```diff
@@ -0,0 +1,101 @@
+import React, {useMemo, useState} from 'react';
+
+import {Select} from 'antd';
+
+import {ExecutorIcon} from '@atoms';
+
+import {Button, Modal, Text} from '@custom-antd';
+
+import usePressEnter from '@hooks/usePressEnter';
+
+import {TestSuiteStepTest} from '@models/test';
+
+import {useAppSelector} from '@redux/hooks';
+import {selectExecutors} from '@redux/reducers/executorsSlice';
+import {getTestExecutorIcon} from '@redux/utils/executorIcon';
+
+import {useGetAllTestsQuery} from '@services/tests';
+
+import {StyledDelayModalContent, StyledOptionWrapper} from '../SettingsTests.styled';
+
+type TestModalProps = {
+  visible: boolean;
+  onClose: () => void;
+  onSubmit: (test: {name: string; type: string}) => void;
+};
+const {Option} = Select;
+
+const TestModal: React.FC<TestModalProps> = props => {
+  const {visible, onClose, onSubmit} = props;
+
+  const {data: allTestsList} = useGetAllTestsQuery();
+
+  const executors = useAppSelector(selectExecutors);
+
+  const [testValue, setTestValue] = useState<string | null>(null);
+
+  const onEvent = usePressEnter();
+
+  const onConfirm = () => {
+    const testData = allTestsData.find(x => x.name === testValue!);
+    if (testData) {
+      onSubmit({name: testData.name, type: testData.type!});
```

Comment:
Is the `type` actually needed? Maybe `namespace` was expected?

Reply:
type is needed for icon